src/core/classes/gas-cost-estimator.ts

```diff
@@ -39,6 +39,31 @@ export class GasCostEstimator<TContract extends BaseContract> {
     return ethers.utils.formatEther(gasUnits.mul(price));
   }
 
+  /**
+   * Estimates the gas limit of a transaction
+   * Pass in the same parameters as the contract's function.
+   * @remarks Estimates the gas limit of a transaction
+   * @example
+   * ```javascript
+   * const gasLimitOfClaim = await nftDrop?.estimator.gasLimitOf("claim", [
+   *   "0x...", // receiver
+   *   1, // quantity
+   *   "0x...", // currency
+   *   1, // price per token
+   *   [], // proofs
+   *   1, // proof max quantity per transaction
+   * ]);
+   * ```
+   * @returns the estimated gas limit in GWEI
+   * @public
+   */
+  public async gasLimitOf(
+    fn: keyof TContract["functions"] | (string & {}),
+    args: Parameters<TContract["functions"][typeof fn]> | any[],
+  ): Promise<BigNumber> {
+    return this.contractWrapper.estimateGas(fn, args);
```

Comment:
can you write a quick test for this for sanity? can just check that gasLimit > 0